.github/workflows/ci.yml

```diff
@@ -24,11 +25,32 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      - name: Set up JDK
-        uses: actions/setup-java@v3
+      - name: Setup up JDK (Java 11)
+        id: setup-java-11
+        if: matrix.java == '11'
+        uses: actions/setup-java@v4
         with:
           distribution: adopt
           java-version: 11
+          cache: sbt
+
+      - name: Setup up JDK (Java 17)
+        id: setup-java-17
+        if: matrix.java == '17'
+        uses: actions/setup-java@v4
+        with:
+          distribution: adopt
+          java-version: 17
```

Comment:
Can we refer to `matrix.java` here rather than having a step for each Java version?

Reply:
I don't think so. I tried that in f7e15d717ef9aa7627f64576d5aab8d6eb0401bd and the CI failed to parse it (I force pushed over it to avoid a dirty history).

So I copied this approach from cats (not sure why they need all those `sbt +update`): https://github.com/typelevel/cats/blob/4f15360e99c27338859d7f0e413d2a38b8ca1f97/.github/workflows/ci.yml#L59-L96